packages/xo-server/src/api/vif.mjs

```diff
@@ -102,6 +108,55 @@ export async function set({
 
     const xapi = this.getXapi(vif)
 
+    // If only the network is changing (not MAC), try to use VIF.move which is non-destructive
+    // VIF.move was introduced in XenServer 7.1 and keeps the VIF UUID, avoiding network downtime
+    if (isNetworkChanged && mac === undefined) {
+      try {
+        // Try VIF.move first (XenServer 7.1+)
+        await xapi.moveVif(vif._xapiId, networkId)
```

Comment:
I think you should call directly `VIF.move` here, since you already have the  full network and vif references 
If you prefer to go through the helper road, I think the best place would be in @xen-orchestra/xapi/vif.mjs , this method will be exposed as xapi.VIF_move